proposals/0201-high-level-interface-overlay-controllers.md

```diff
@@ -139,15 +142,58 @@ typedef void (^SDLOverlayControllerCompletionHandler)(SDLResult result, NSError
 
 Every overlay comes with a duration that specifies how long the view should be active on the HMI (visible on the screen). This parameter maps to all the different names used for duration such as `duration` (`Alert`), `timeout` (`Slider`, `PerformInteraction` and `ScrollableMesasge`) and `maxDuration` (`PerformAudioPassThru`).
 
-Depending on the subclass (and the min/max values of the parameter) the duration value will be ignored if it exceeds the value range. 
-1. If `duration` value is greater than the max value, the RPC parameter should be set to max value. 
+Depending on the subclass (and the min/max values of the parameter) the duration value will be ignored if it exceeds the value range.
+1. If `duration` value is greater than the max value, the RPC parameter should be set to max value.
 2. If `duration` value is smaller than the min value, the RPC parameter should be set to min value.
 3. If `duration` value is set to 0 and the RPC parameter is optional, the parameter should be omitted.
 4. If `duration` value is set to 0 and the RPC parameter is required, the parameter should be set to max value (see `PerformAudioPassThru.maxDuration`).
 
-#### Main screen overlay
+### Main screen overlay
+
+`SDLOverlayController` inherits `SDLViewController`. With this, it is possible to manipulate the app's main screen during a presentation. This can be very useful to:
+- show a text message for "Loading..." or "Please wait" if the overlay takes more time to load
+- add buttons to cancel the overlay request if possible.
+
+This feature is planned in the proposal for Overlay controller related to choice sets.
+
+To realize above behavior, the app developer must create a new view and place it in the `view` property of the overlay controller. When presenting the overlay controller with a view object the view controller manager would override the full screen of the app but it may be handy to just override a single text field. With that a new property should be added to `SDLView` for an opaque mode.
+
+```objc
+@interface SDLView
+
+/**
+ * Specifies whether the view is opaque. An opaque view will be presented on the app's screen
+ * overriding all content from underlying views. If this property is set to NO the view is
+ * transparent and the view controller manager will composite content from underlying views.
+ * The default value is YES.
+ */
+@property (nonatomic, assign) BOOL opaque;
+
+@end
+```
+
+If the `opaque` flag is set to NO the view will be called transparent. It should allow views or view parts from the underlying view controller to stay visible while the overlay controller is presenting. The opaque mode is view type related and allows . With this flag:
```

Comment:
Is there additional information missing here: `type related and allows .` ?

Update to:
If the `opaque` flag is set to NO the view will be called transparent. It should allow views or view parts from the underlying view controller to stay visible while the overlay controller is presenting. The opaque mode **is a** view **type related and allows .** With this flag:

Reply:
> Is there additional information missing here: `type related and allows .` ?

oh wow... sorry for that